BTrees/_base.py

```diff
@@ -113,7 +113,7 @@ def _search(self, key):
             k = keys[i]
             if k is key or k == key:
                 return i
-            if k < key:
+            if k is None or (key is not None and k < key):
                 low = i + 1
```

Comment:
I think the `key is not None` part of this is an unnecessary extra comparison, since `or` short-circuits.